engine/src/exec-files.cpp

```diff
@@ -1953,8 +1953,19 @@ static void MCFilesReadComplete(MCExecContext& ctxt, MCValueRef p_output, IO_sta
         if (t_textmode)
         {
             MCAutoStringRef t_output;
-            /* UNCHECKED*/ MCStringConvertLineEndingsToLiveCode((MCStringRef)p_output, &t_output);
-            ctxt . SetItToValue(*t_output);
+            if (!MCStringNormalizeLineEndings((MCStringRef)p_output, 
+                                              kMCStringLineEndingStyleLF, 
+                                              true, 
+                                              &t_output, 
+                                              nullptr))
+            {
+                // handle error condition?
```

Comment:
When a foundation function returns false then it will have set a libfoundation error at some point (almost all foundation functions only fail due to out-of-memory); so for normal Exec type functions, you would do `{ ctxt.Throw(); return }`. However, that depends on whether the caller is expecting an error from this function - which it doesn't look like it is. In this case, you'd probably be fine just using the same result as IO_ERROR.

Checking return values on foundation functions is more about cleanliness than anything else - they usually only fail due to OOM, which is generally uncatchable anyway (due to over-commit policies on modern OSes).